avr-hal-generic/src/wdt.rs

```diff
@@ -0,0 +1,141 @@
+//! WDT Implementation
+
+pub use embedded_hal::watchdog;
+
+/// Approximate length of the time-out period before the watchdog provides a system reset.
+///
+/// After enabling the watchdog timer, call [`feed`](trait.Watchdog.html#tymethod.feed) before the
+/// period ends to prevent a reset.
+///
+/// Some AVR models do not support all period lengths.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum WatchdogTimeOutPeriod {
+    /// 16 milliseconds
+    Ms16,
+    /// 32 milliseconds
+    Ms32,
+    /// 64 milliseconds
+    Ms64,
+    /// 125 milliseconds
+    Ms125,
+    /// 250 milliseconds
+    Ms250,
+    /// 500 milliseconds
+    Ms500,
+    /// 1 second
+    Ms1000,
+    /// 2 seconds
+    Ms2000,
+    /// 4 seconds
+    Ms4000,
+    /// 8 seconds
+    Ms8000,
+}
+
+/// Implement traits for the watchdog interface
+#[macro_export]
+macro_rules! impl_wdt {
+    (
+        $(#[$wdt_attr:meta])*
+        pub struct $Wdt:ident {
+            mcu_status_register: $MCUSR:ty,
+            peripheral: $WDT:ty,
+        }
+    ) => {
+        use avr_device::generic::Reg;
+        pub use avr_hal::hal::watchdog::*;
```

Comment:
This depends on the crate being imported as `avr_hal`.  In macros, one should usually refer to the own crate using `$crate` as that is guaranteed to always work.

That said, I think the imports heres should be removed; they would make it impossible to use this macro in a module that for some reason has these items already imported.  Instead, inside the macro, I suggest referring to everything by its full path.  So further down, e.g. refer to `WatchdogEnable` like this:

```rust
impl $crate::hal::watchdog::WatchdogEnable for $Wdt {
```

Reply:
(And `Reg` should no longer be necessary with the other suggested changes.)